AsyncDisplayKit/Private/_ASHierarchyChangeSet.m

```diff
@@ -314,8 +314,10 @@ + (void)sortAndCoalesceChanges:(NSMutableArray *)changes
   NSMutableIndexSet *allIndexes = [NSMutableIndexSet new];
   
   for (_ASHierarchySectionChange *change in changes) {
-    [change.indexSet enumerateIndexesUsingBlock:^(NSUInteger idx, __unused BOOL *stop) {
-      animationOptions[@(idx)] = @(change.animationOptions);
+    [change.indexSet enumerateRangesUsingBlock:^(NSRange range, BOOL * _Nonnull stop) {
+      for (NSUInteger i = range.location; i < NSMaxRange(range); i++) {
+        animationOptions[@(i)] = @(change.animationOptions);
```

Comment:
What about changing animationOptions to a std::unordered_map? Do we expose it somehow?